camelot/view/controls/delegates/customdelegate.py

```diff
@@ -218,13 +220,24 @@ def setEditorData(self, editor, index):
         editor.set_value(value)
 
         # update actions
-        action_states = index.model().data(index, ActionStatesRole)
-        if action_states is None:
-            return
-        for action_widget in editor.findChildren(ActionToolbutton):
-            if action_widget.action_route in action_states:
-                state = action_states[action_widget.action_route]
-                action_widget.set_state_v2(state)
+        if isinstance(editor, (Many2OneEditor, ChoicesEditor, DbImageEditor, FloatEditor, FileEditor)):
```

Comment:
het is niet de bedoeling dat er in de delegate code 'discriminatie' is op basis vh editor type.

dit gezien elke editor eigenlijk al een specifieke delegate heeft.

ik zou dus zeggen, steek de generieke code voor 'alle' editors in een method van de custom delegate (dat mag ook voor de editors waarvoor er nu geen actions zijn), en overschrijf die method in de One2ManyDelegate